ee/app/livechat-enterprise/server/lib/VisitorInactivityMonitor.js

```diff
@@ -0,0 +1,65 @@
+import { SyncedCron } from 'meteor/littledata:synced-cron';
+import { TAPi18n } from 'meteor/rocketchat:tap-i18n';
+
+import { settings } from '../../../../../app/settings/server';
+import { LivechatRooms, LivechatDepartment } from '../../../../../app/models/server';
+import { Livechat } from '../../../../../app/livechat/server/lib/Livechat';
+
+export class VisitorInactivityMonitor {
+	constructor() {
+		this._started = false;
+		this._name = 'Omnichannel Visitor Inactivity Monitor';
+	}
+
+	start() {
+		this._startMonitoring();
+	}
+
+	_startMonitoring() {
+		if (this.isRunning()) {
+			return;
+		}
+		const everyMinute = '* * * * *';
+		SyncedCron.add({
+			name: this._name,
+			schedule: (parser) => parser.cron(everyMinute),
+			job: () => {
+				this.handleAbandonedRooms();
+			},
+		});
+		this._started = true;
+	}
+
+	stop() {
+		if (!this.isRunning()) {
+			return;
+		}
+
+		SyncedCron.remove(this._name);
+
+		this._started = false;
+	}
+
+	isRunning() {
+		return this._started;
+	}
+
+	closeRooms(room) {
+		let comment = settings.get('Livechat_abandoned_rooms_closed_custom_message') || TAPi18n.__('Closed_automatically');
+		if (room.departmentId) {
+			const department = LivechatDepartment.findOneById(room.departmentId);
+			comment = department.abandonedRoomsCloseCustomMessage || comment;
+		}
+		Livechat.closeRoom({
```

Comment:
The user who is closing the chats is required here. So, my suggestion is to use `rocket.cat` to as a user to perform this action. 
Then you don't need to set `system` here:
https://github.com/RocketChat/Rocket.Chat/pull/17055/files#diff-3e725e94e4a647e422a10d903b96cf59R349